org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/ProjectCommand.java

```diff
@@ -270,14 +271,15 @@ public static IJavaProject getJavaProjectFromUri(String uri) throws CoreExceptio
 		}
 
 		// For multi-module scenario
+		IPath inputPath = FileUtil.toPath(new URI(uri));
 		Arrays.sort(containers, (Comparator<IContainer>) (IContainer a, IContainer b) -> {
-			return a.getFullPath().toPortableString().length() - b.getFullPath().toPortableString().length();
+			return inputPath.makeRelativeTo(a.getLocation()).segmentCount() - inputPath.makeRelativeTo(b.getLocation()).segmentCount();
```

Comment:
![捕获](https://user-images.githubusercontent.com/6193897/131969594-a7fc9225-b6a5-4640-b26b-ed843122219d.PNG)

For the case above, you just need to use `a.getFullPath().segmentCount() - b.getFullPath().segmentCount()`. This will put the directly belonged project in front.

Reply:
make sense.